src/components/OTP.vue

```diff
@@ -0,0 +1,308 @@
+    inputBoxPlaceholderText() {
+      return this.programData["text"]["default"]["placeholder"];
+    },
+
+    /** Checks if any userID has been entered */
+    isAnyUserIDPresent() {
+      return this.userIDList != undefined && this.userId != "";
+    },
+
+    /** Whether the 'Request OTP' button should be disabled */
+    isRequestOTPButtonDisabled() {
+      return (
+        this.userId == undefined ||
+        this.invalidInputMessage != "" ||
+        this.isOTPSent == true
+      );
+    },
+
+    /**
+     * Whether the submit button is disabled
+     * Returns true if any of the following conditions are met:
+     * - if 'Request OTP' button is enabled
+     * - if OTP is being typed
+     */
+    isSubmitButtonDisabled() {
+      return !(this.isRequestOTPButtonDisabled || this.OTPCode.length > 0);
+    },
+
+    /** Checks if the current input entry has the required number of characters */
+    isCurrentEntryIncomplete() {
+      return this.userId.length < this.programData["input"]["maxLengthOfId"];
+    },
+
+    /** Whether input being typed is in the correct format */
+    isInvalidInputMessageShown() {
+      return this.invalidInputMessage != null;
+    },
+  },
+  methods: {
+    /** Determines how the input box should look.
+     * - If an input error needs to be displayed, the box has a a red border.
+     * - Otherwise, it has an opacity of 30.
+     * @param {Number} index - index of the input box
+     */
+    selectInputBoxClasses() {
+      return [
+        {
+          "border-red-600 focus:border-red-600": this.invalidInputMessage,
+          "pointer-events-none opacity-30": !this.invalidInputMessage,
+        },
+      ];
+    },
+
+    /** Calls the mapping function to validate the typed character
+     * @param {Object} event - event triggered when a character is typed
+     */
+    isValidEntry(event) {
+      if (
+        validationTypeToFunctionMap[this.programData["input"]["basicValidationType"]](
+          event
+        )
+      ) {
+        return true;
+      } else event.preventDefault();
+    },
+
+    /** Resets the invalid input message */
+    resetInvalidInputMessage() {
+      this.invalidInputMessage = "";
+    },
+
+    /** This function is called whenever something is entered in the input box.
+     * It checks if the required number of characters are being typed.
+     * @param {Object} event - the event which triggered this function
+     * @param {Number} index - the index of the input field
+     */
+    updateUserId(event) {
+      if (event.target.value.length == 0) {
+        this.invalidInputMessage = "";
+      } else if (event.target.value.length < this.programData["input"]["maxLengthOfId"]) {
+        this.invalidInputMessage = this.programData["text"]["default"]["invalid"][
+          "input"
+        ];
+      } else {
+        this.resetInvalidInputMessage();
+      }
+      if (event.target.value.length > this.programData["input"]["maxLengthOfId"]) {
```

Comment:
Why not add this condition as another else if attached to the above if/else logic?